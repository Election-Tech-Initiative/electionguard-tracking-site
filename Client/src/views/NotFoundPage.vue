<template>
  <div class="section">
    <div class="container">
      <div class="content is-size-3">
        <i18n path="notFoundMessage">
          <span class="is-inline-block has-text-weight-bold">{{ $t('notFound') }}</span>
        </i18n>
      </div>
      <div class="content">        
        <router-link class="button is-dark" to='/'>{{ $t('returnHome') }}</router-link>    
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class NotFoundPage extends Vue {
}
</script>

<style>

</style>
