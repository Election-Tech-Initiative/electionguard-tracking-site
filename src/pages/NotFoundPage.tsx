import React from 'react';
import Title from '../components/Title';

const notFound = 'Not Found';

export interface NotFoundPageProps {}

const NotFoundPage: React.FunctionComponent<NotFoundPageProps> = () => <Title title={`404 ${notFound}`} />;

export default NotFoundPage;
