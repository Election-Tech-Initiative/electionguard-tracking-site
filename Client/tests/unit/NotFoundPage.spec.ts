import { mount, RouterLinkStub } from '@vue/test-utils';

import NotFoundPage from '@/views/NotFoundPage.vue';

describe('NotFoundPage.vue', () => {
    it('renders', () => {
        const wrapper = mount(NotFoundPage, {
            mocks: {
                $t: (msg: string): string => msg
            },
            stubs: {
                i18n: true,
                RouterLink: RouterLinkStub,
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
