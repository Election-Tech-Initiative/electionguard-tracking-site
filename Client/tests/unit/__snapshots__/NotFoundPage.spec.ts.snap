// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage.vue renders 1`] = `
<div
  class="section"
>
  <div
    class="container"
  >
    <div
      class="content is-size-3"
    >
      <i18n-stub
        path="notFoundMessage"
      >
        <span
          class="is-inline-block has-text-weight-bold"
        >
          notFound
        </span>
      </i18n-stub>
    </div>
     
    <div
      class="content"
    >
      <a
        class="button is-dark"
      >
        returnHome
      </a>
    </div>
  </div>
</div>
`;
